app/src/main/java/org/mozilla/fenix/onboarding/view/NotificationPermissionDialogScreen.kt

```diff
@@ -0,0 +1,221 @@
+ */
+@Composable
+fun NotificationPermissionDialogScreen(
+    onDismiss: () -> Unit,
+    grantNotificationPermission: () -> Unit,
+) {
+    NotificationPermissionContent(
+        notificationPermissionPageState = NotificationPageState,
+        onDismiss = onDismiss,
+        onPrimaryButtonClick = grantNotificationPermission,
+        onSecondaryButtonClick = onDismiss,
+    )
+}
+
+@Composable
+private fun NotificationPermissionContent(
+    notificationPermissionPageState: NotificationPermissionPageState,
+    onDismiss: () -> Unit,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    BoxWithConstraints(Modifier.fillMaxSize()) {
+        val boxWithConstraintsScope = this
+        Column(
+            modifier = modifier
+                .background(FirefoxTheme.colors.layer1)
+                .fillMaxSize()
+                .padding(bottom = 32.dp)
+                .statusBarsPadding()
+                .navigationBarsPadding(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+            IconButton(
+                onClick = onDismiss,
+                modifier = Modifier.align(Alignment.End),
+            ) {
+                Icon(
+                    painter = painterResource(id = R.drawable.mozac_ic_close),
+                    contentDescription = stringResource(R.string.content_description_close_button),
+                    tint = FirefoxTheme.colors.iconPrimary,
+                )
+            }
+
+            NotificationPermissionPage(
+                pageState = notificationPermissionPageState,
+                onPrimaryButtonClick = onPrimaryButtonClick,
+                onSecondaryButtonClick = onSecondaryButtonClick,
+                modifier = Modifier
+                    .weight(1f)
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp)
+                    .verticalScroll(rememberScrollState()),
+                imageModifier = Modifier
+                    .height(boxWithConstraintsScope.maxHeight.times(IMAGE_HEIGHT_RATIO)),
+            )
+        }
+    }
+}
+
+/**
+ * A page for displaying Notification Permission Content.
+ *
+ * @param pageState The page content that's displayed.
+ * @param onPrimaryButtonClick Invoked when the user clicks the primary button.
+ * @param onSecondaryButtonClick Invoked when the user clicks the secondary button.
+ * @param modifier The modifier to be applied to the Composable.
+ */
+@Composable
+private fun NotificationPermissionPage(
+    pageState: NotificationPermissionPageState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    imageModifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier,
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.SpaceEvenly,
```

Comment:
Looking at the layout, I feel like the space between the title and the image is a bit far comparing to the design.  With this [layout.patch](https://github.com/mozilla-mobile/fenix/files/10395946/layout.patch)  it seems to match it better.  Does this looks good to you?
Before(Left), After(Right)
![BeforeAfter](https://user-images.githubusercontent.com/6579541/211916903-f063cdad-c97f-41c1-9e0b-3eee2eade353.jpg)

Reply:
@Kate-Galetski which one do you think is the one to go with? Thanks

Reply:
As confirmed with @Kate-Galetski that the after is not better. I'll mark this suggestion as resolved.